modules/markup/html.go

```diff
@@ -359,6 +359,10 @@ func visitNode(ctx *RenderContext, procs, textProcs []processor, node *html.Node
 		if attr.Key == "id" && !(strings.HasPrefix(attr.Val, "user-content-") || blackfridayExtRegex.MatchString(attr.Val)) {
 			node.Attr[idx].Val = "user-content-" + attr.Val
 		}
+		// Add user-content- to # links too
+		if attr.Key == "href" && strings.HasPrefix(attr.Val, "#") && !(strings.HasPrefix(attr.Val, "#user-content-") || blackfridayExtRegex.MatchString(strings.TrimPrefix(attr.Val, "#"))) {
```

Comment:
Hmm… I'm wondering if we should extract everything after the first `&&` until the `{` into a separate method.

Reply:
Refactoring suggestion (untested):

```js
for idx, attr := range node.Attr {
	const val = strings.TrimPrefix(attr.Val, "#");
	const shouldHavePrefix = !(strings.HasPrefix(val, "user-content-") || blackfridayExtRegex.MatchString(val));

	if attr.Key == "id" && shouldHavePrefix) {
		node.Attr[idx].Val = "user-content-" + attr.Val
	}
	// Add user-content- to # links too
	if attr.Key == "href" && strings.HasPrefix(attr.Val, "#") && shouldHavePrefix) {
		node.Attr[idx].Val = "#user-content-" + val
	}

	if attr.Key == "class" && attr.Val == "emoji" {
		textProcs = nil
	}
}

```